importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -4575,6 +4575,25 @@ Note* MusicXMLParserPass2::note(const QString& partId,
             setNoteHead(note, noteheadColor, noteheadParentheses, noteheadFilled);
             note->setVisible(printObject); // TODO also set the stem to invisible
 
+            if (mnd.calculatedDuration().isValid()
+                && mnd.specifiedDuration().isValid()
+                && mnd.calculatedDuration().isNotZero()
+                && mnd.calculatedDuration() != mnd.specifiedDuration()) {
+
+                  // convert duration into note length
+                  Fraction durationMult { (mnd.specifiedDuration() / mnd.calculatedDuration()).reduced() };
+                  durationMult = (1000 * durationMult).reduced();
+                  const int noteLen = durationMult.numerator() / durationMult.denominator();
+
+                  NoteEventList nel;
+                  NoteEvent ne;
+                  ne.setLen(noteLen);
+                  nel.append(ne);
+                  note->setPlayEvents(nel);
+                  if (c)
+                        c->setPlayEventType(PlayEventType::User);
+                  }
+
```

Comment:
This change here somwhat collide with changes from #8581, resuls in mtest failures:
```
62/62 Test #51: tst_mxml_io ......................***Failed   77.27 sec
--- /home/runner/work/MuseScore/MuseScore/mtest/musicxml/io/testCueGraceNotes_ref.mscx	2021-08-03 11:40:32.492648198 +0000
+++ testCueGraceNotes.mscx	2021-08-03 12:05:19.714046284 +0000
@@ -153,6 +153,11 @@
             <durationType>eighth</durationType>
             <acciaccatura/>
Errors while running CTest
             <Note>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>65</pitch>
               <tpc>13</tpc>
               <play>0</play>
@@ -233,6 +238,11 @@
             <durationType>eighth</durationType>
             <acciaccatura/>
             <Note>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>69</pitch>
               <tpc>17</tpc>
               <play>0</play>
@@ -300,6 +310,11 @@
             <durationType>eighth</durationType>
             <acciaccatura/>
             <Note>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>69</pitch>
               <tpc>17</tpc>
               <play>0</play>
@@ -386,6 +401,11 @@
             <durationType>eighth</durationType>
             <acciaccatura/>
             <Note>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>67</pitch>
               <tpc>15</tpc>
               </Note>
@@ -449,6 +469,11 @@
             <durationType>eighth</durationType>
             <appoggiatura/>
             <Note>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>60</pitch>
               <tpc>14</tpc>
               </Note>
@@ -493,6 +518,11 @@
               <Accidental>
                 <subtype>accidentalNatural</subtype>
                 </Accidental>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>74</pitch>
               <tpc>16</tpc>
               </Note>
@@ -645,6 +675,11 @@
               <Accidental>
                 <subtype>accidentalFlat</subtype>
                 </Accidental>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>70</pitch>
               <tpc>12</tpc>
               <play>0</play>
@@ -743,6 +778,11 @@
               <Accidental>
                 <subtype>accidentalSharp</subtype>
                 </Accidental>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>66</pitch>
               <tpc>20</tpc>
               <play>0</play>
@@ -807,6 +847,11 @@
             <durationType>eighth</durationType>
             <acciaccatura/>
             <Note>
+              <Events>
+                <Event>
+                  <len>0</len>
+                  </Event>
+                </Events>
               <pitch>65</pitch>
               <tpc>13</tpc>
               <play>0</play>
   <diff -u /home/runner/work/MuseScore/MuseScore/mtest/musicxml/io/testCueGraceNotes_ref.mscx testCueGraceNotes.mscx failed--- 
```

Reply:
Maybe @iveshenry18 can chip in here?

Reply:
This would affect #8429 too, once #8581,  gets ported to master?

Reply:
The conflict with #8581 seems like it would be relatively simple to reconcile when porting to master; however, I also suspect that there would be a more significant conflict with an updated tuplet-rounding system I wrote in #8766

Reply:
So we have to align on how to proceed. I would expect we will finally want to have master contain the sum of all mentioned PRs. I.e. we'l simply have to fix the conflicts.

Reply:
And as the tuplet handling code (and especially the error handling) is not very good, implementing the updated tuplet-rounding system should help. I would like to review that later, as I cannot quickly judge the impact.

Reply:
Sounds good!